src/js/components/Carousel/Carousel.js

```diff
@@ -1,242 +1,213 @@
 import React, {
   Children,
-  useCallback,
   useContext,
-  useEffect,
   useState,
-  useRef,
+  useEffect,
+  useCallback,
 } from 'react';
-
-import { normalizeColor } from '../../utils';
-import { defaultProps } from '../../default-props';
-import { ThemeContext } from '../../contexts';
-
-import { Box } from '../Box';
-import { Button } from '../Button';
-import { Keyboard } from '../Keyboard';
-import { Stack } from '../Stack';
-
 import { CarouselChild } from './CarouselChild';
+import { CarouselControls } from './CarouselControls';
+import { StyledCarouselContainer } from './StyledCarousel';
 import { CarouselPropTypes } from './propTypes';
+import { Keyboard } from '../Keyboard';
+import { ThemeContext } from '../../contexts';
+import { defaultProps } from '../../default-props';
 
 const Carousel = ({
   activeChild,
-  initialChild,
-  onChild,
-  play,
+  continuous,
   children,
   controls,
+  height,
   fill,
-  onFocus,
-  onBlur,
+  width,
+  initialChild,
+  onChild,
+  play,
   ...rest
 }) => {
+  const containerRef = React.createRef();
   const theme = useContext(ThemeContext) || defaultProps.theme;
-  const [focus, setFocus] = useState();
-  const timerRef = useRef();
-
-  const [indexes, setIndexes] = useState({
-    activeIndex: activeChild !== undefined ? activeChild : initialChild,
+  const numSlides = children.length;
+  const noContainer = !fill && (!height || !width);
+  const animationDuration =
+    play && play < theme.carousel.animation.duration
+      ? play
+      : theme.carousel.animation.duration;
+
+  const [current, setCurrent] = useState(activeChild || initialChild);
+  const [previous, setPrevious] = useState(undefined);
+  const [activeChildState, setActiveChildState] = useState(activeChild);
+  const [direction, setDirection] = useState(undefined);
+  const [inTransition, setInTransition] = useState(false);
+  const [containerProps, setContainerProps] = useState({
+    containerHeight: undefined,
```

Comment:
Can these just be called `height` and `width`? Since they are already in `containerProps`, they already have a context for the "container".

Reply:
These props are only meant to be processed in the styled component. Since it is a styled `Box`, if we name these `height` and `width` then the values will be overwritten.

Reply:
I think Box can handle us passing an explicit height and width already.